src/Entity/CitizenProject.php

```diff
@@ -319,6 +320,12 @@ public function approved(string $timestamp = 'now'): void
     public function update(string $name, NullablePostAddress $address): void
     {
         $this->setName($name);
+        $this->setSubtitle($subtitle);
+        $this->category = $category;
+        $this->setAssistanceNeeded($assistanceNeeded);
```

Comment:
Inside an Entity, try to avoid setter and use attribute instead

Reply:
I disagree with you. Doing so would break the encapsulation principle.

Reply:
You are inside the object, the encapsulation principle is already broken.

Reply:
No, if one extends the behavior of the class by any way: inheritance, decorator, etc. This would break encapsulation. It's related to what it is explained in this article -> https://ocramius.github.io/blog/fluent-interfaces-are-evil/